scarb/src/bin/scarb/commands/proc_macro_server.rs

```diff
@@ -1,7 +1,52 @@
 use anyhow::Result;
-use scarb::core::Config;
+use scarb::{
+    compiler::{plugin::proc_macro::ProcMacroHost, CairoCompilationUnit, CompilationUnit},
+    core::{Config, Workspace},
+    ops::{self, FeaturesOpts, FeaturesSelector},
+};
 
 #[tracing::instrument(skip_all, level = "info")]
-pub fn run(_config: &Config) -> Result<()> {
-    unimplemented!()
+pub fn run(config: &mut Config) -> Result<()> {
+    let ws = ops::read_workspace(config.manifest_path(), config)?;
+    let resolve = ops::resolve_workspace(&ws)?;
+    let compilation_units = ops::generate_compilation_units(
+        &resolve,
+        &FeaturesOpts {
+            features: FeaturesSelector::AllFeatures,
+            no_default_features: false,
+        },
+        &ws,
+    )?;
+
+    for unit in &compilation_units {
+        if let CompilationUnit::ProcMacro(_) = unit {
+            ops::compile_unit(unit.clone(), &ws)?;
```

Comment:
why are you compiling everything here?

Reply:
Not sure what could be omitted. If you have better inside can you type here what could be removed?

Reply:
Compiling creates the `RootDatabase`, runs full compilation and outputs artefacts. That's nothing we can make use of, so we are safe to skip this.

The rest of the code here all looks okay for me here. We basically need to `scarb fetch` before booting up, so you just need to:
1. Read the workspace
2. Resolve it, including downloading all dependencies.
3. Generate compilation units to gather all compiler db parameters.
4. Create proc macro host.

Reply:
@mkaput but those are only macros -> per match one line prior

Reply:
> @mkaput but those are only macros -> per match one line prior

I don't understand, could you elaborate please?

Reply:
```
        if let CompilationUnit::ProcMacro(_) = unit {
            ops::compile_unit(unit.clone(), &ws)?;
```
Unit is only compiled, if it fulfills the `let CompilationUnit::ProcMacro(_) = unit` match, thus is guaranteed to represent a procedural macro.

Reply:
🤦🏻 I didn't notice this. @Draggu Could you please add a comment elaborating that we are compiling procedural macros here?